server/src/com/cloud/api/ApiResponseHelper.java

```diff
@@ -3461,10 +3470,21 @@ public NicResponse createNicResponse(Nic result) {
         response.setNetmask(result.getIPv4Netmask());
         response.setMacAddress(result.getMacAddress());
 
+        if (result.getBroadcastUri() != null) {
+            response.setBroadcastUri(result.getBroadcastUri().toString());
```

Comment:
@ustcweizhou As per api doc these info should be part of response. Even listVirtualMachines with details=nics 
`command=listVirtualMachines&details=nics&id=6eea5ceb-e943-4ef3-9f40-4671364d03ab&response=json` provides these details.